catalog/core/catalog-core-transform-impl/src/test/java/org/codice/ddf/catalog/transform/impl/TransformImplTest.java

```diff
@@ -0,0 +1,497 @@
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.QueryResponseTransformer;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.codice.ddf.catalog.locator.TransformerLocator;
+import org.codice.ddf.catalog.transform.MultiInputTransformer;
+import org.codice.ddf.catalog.transform.MultiMetacardTransformer;
+import org.codice.ddf.catalog.transform.Transform;
+import org.codice.ddf.catalog.transform.TransformResponse;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+
+@SuppressWarnings("unchecked")
+public class TransformImplTest {
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private File tempFile;
+
+  private Metacard metacard;
+
+  private TransformerLocator transformerLocator;
+
+  private Transform transform;
+
+  private Supplier<String> idSupplier;
+
+  @Before
+  public void setup() throws IOException {
+    tempFile = temporaryFolder.newFile("tempFile");
+    metacard = mock(Metacard.class);
+    transformerLocator = mock(TransformerLocator.class);
+    UuidGenerator uuidGenerator = mock(UuidGenerator.class);
+    transform = new TransformImpl(transformerLocator, uuidGenerator);
+    idSupplier = mock(Supplier.class);
+    when(idSupplier.get()).thenReturn(null);
+  }
+
+  @Test
+  public void testTransformSourceResponseByTransformerId() throws CatalogTransformerException {
+
+    BinaryContent mockBinaryContent = mock(BinaryContent.class);
+
+    QueryResponseTransformer transformer = mock(QueryResponseTransformer.class);
+    when(transformer.transform(any(SourceResponse.class), any(Map.class)))
+        .thenReturn(mockBinaryContent);
+
+    when(transformerLocator.findQueryResponseTransformers(eq("xyz")))
+        .thenReturn(Collections.singletonList(transformer));
+
+    BinaryContent binaryContent =
+        transform.transform(mock(SourceResponse.class), "xyz", Collections.emptyMap());
+
+    assertThat(binaryContent, is(mockBinaryContent));
+  }
+
+  @Test
+  public void testTransformSourceResponseByMimeType()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    BinaryContent mockBinaryContent = mock(BinaryContent.class);
+
+    QueryResponseTransformer transformer = mock(QueryResponseTransformer.class);
+    when(transformer.transform(any(SourceResponse.class), any(Map.class)))
+        .thenReturn(mockBinaryContent);
+
+    when(transformerLocator.findQueryResponseTransformers(any(MimeType.class)))
```

Comment:
Shouldn't this `MimeType` matcher be `eq(new MimeType("text/xml"))`?

Reply:
fixed with a custom argument matcher